test/unit/report_importer_test.rb

```diff
@@ -10,7 +10,7 @@ class ReportImporterTest < ActiveSupport::TestCase
   end
 
   test 'it should import reports with no metrics' do
-    r = ReportImporter.import(read_json_fixture('report-empty.json'))
+    r = ConfigReportImporter.import(read_json_fixture('report-empty.json'))
     assert r
     assert_equal({}, r.metrics)
```

Comment:
I think this should have caught the problem in the metrics method when none were supplied, perhaps *because* it was using the ConfigReportImporter instead of ReportImporter.  It's not testing at the right level.